### ISSUE TITLE
Client fails to load data

### ISSUE THREAD

**Issue by dsmmcken** [deephaven/deephaven-core#2778]
**Steps to reproduce**
On the table "Active Streaks", on https://beat-the-streak.deephaven.io/ide/  if you scroll all the way to the right, then scrub the scrollbar down and back to the top for a bit it fails with this error. Haven't been able to reproduce any other way.

From @niloc132 
> it appears that the server is saying "here's a column, it has 26 values and no nulls. Here's all zero values."
  and the client gets grumpy that in fact 0 < 26
  specifically, the server says "here's zero bytes, which you can read", which the client says "26 * 4 is 104, which is greater than 0"

```
dh-internal.js:1 Uncaught Error: java.lang.AssertionError
    at AssertionError.createError (dh-core.js:1189:10)
    at AssertionError.initializeBackingError (dh-core.js:1209:46)
    at AssertionError.Throwable (dh-core.js:1147:8)
    at AssertionError.Error_0 (dh-core.js:32482:16)
    at new AssertionError (dh-core.js:32492:14)
    at Object.readArrowBuffer (dh-core.js:16189:20)
    at Object.createSnapshot (dh-core.js:16136:38)
    at Object.$startAndMaybeFlush (dh-core.js:14989:23)
    at Object.$apply_3 (dh-core.js:14978:11)
    at Function.apply_53 (dh-core.js:15007:9)
    at lambda (dh-core.js:149:22)
    at dh-internal.js:1:356460
    at Array.forEach (<anonymous>)
    at onMessage (dh-internal.js:1:356439)
    at dh-internal.js:1:43913
    at Array.forEach (<anonymous>)
    at e.rawOnMessage (dh-internal.js:1:43875)
    at dh-internal.js:1:41688
    at Array.forEach (<anonymous>)
    at e.onTransportChunk (dh-internal.js:1:41563)
    at dh-internal.js:1:48569
10:58:33.451 dh-internal.js:1 Uncaught Error: java.lang.IllegalArgumentException
    at IllegalArgumentException.createError (dh-core.js:1189:10)
    at IllegalArgumentException.initializeBackingError (dh-core.js:1209:46)
    at IllegalArgumentException.Throwable (dh-core.js:1147:8)
    at IllegalArgumentException.Exception (dh-core.js:1221:16)
    at IllegalArgumentException.RuntimeException (dh-core.js:1238:16)
    at new IllegalArgumentException (dh-core.js:1808:23)
    at Object.$position (dh-core.js:34299:16)
    at Object.readArrowBuffer (dh-core.js:16339:22)
    at Object.createSnapshot (dh-core.js:16136:38)
    at Object.$startAndMaybeFlush (dh-core.js:14989:23)
    at Object.$apply_3 (dh-core.js:14978:11)
    at Function.apply_53 (dh-core.js:15007:9)
    at lambda (dh-core.js:149:22)
    at dh-internal.js:1:356460
    at Array.forEach (<anonymous>)
    at onMessage (dh-internal.js:1:356439)
    at dh-internal.js:1:43913
    at Array.forEach (<anonymous>)
    at e.rawOnMessage (dh-internal.js:1:43875)
    at dh-internal.js:1:41688
    at Array.forEach (<anonymous>)
    at e.onTransportChunk (dh-internal.js:1:41563)
    at dh-internal.js:1:48569
    ```

**Comment by niloc132** [comment]
When a column isn't part of a subscription, it is still sent, but as an empty column. When a _boolean_ column isn't part of the subscription, the reinterpret-as-byte wiring isn't kicking in properly, and instead the column is being sent as if it were string values (as the server recognizes an uninterpreted java.lang.Boolean as an object chunk, which isn't one of the few supported Object chunk types). This results in three flight buffers being sent for the empty "serialized as a string" column, instead of the two expected for a boolean column, and the client is off by one when reading.

Any wide enough table such that the UI doesn't display all columns will trigger this - make many columns, append a single boolean column out of sight, and then add a format column for the first column after that, so that the client must correctly decode the format column in order to display the first column, but one too many buffers have already been consumed.

```
from deephaven import *
cols = [f"C{i}=i" for i in range(100)]
cols[98] = "C98 = true"
cols[99] = "C0__WTABLE_FORMAT=`#ff0000`"
t = empty_table(100).update(formulas=cols)
```

**Comment by nbauernfeind** [comment]
This specific issue is fixed in #2799. I opened #2800 and #2801 to capture two additional issues discovered while investigating this issue.
